projects/netgrif-components-core/src/lib/authorization/role/role-guard.service.ts

```diff
@@ -37,22 +37,39 @@ export class RoleGuardService implements CanActivate {
     }
 
     public canAccessView(view: View, url: string): boolean {
-        if (typeof view.access !== 'string' && view.access.hasOwnProperty('role')) {
-            const allowedRoles = this.parseRoleConstraints(view.access.role, url);
-
-            return allowedRoles.some(constraint => {
-                if (constraint.roleIdentifier) {
-                    return this._userService.hasRoleByIdentifier(constraint.roleIdentifier, constraint.processIdentifier);
-                } else {
-                    return this._userService.hasRoleByName(constraint.roleName, constraint.processIdentifier);
-                }
-            });
+        if (typeof view.access !== 'string' && (view.access.hasOwnProperty('role') || view.access.hasOwnProperty('bannedRole'))) {
+
+            if (view.access.hasOwnProperty('bannedRole')) {
+                const bannedRoles = this.parseRoleConstraints(view.access.bannedRole, url);
+
+                return (bannedRoles.length == 0) || (bannedRoles.length > 0 && !bannedRoles.some(constraint => {
+                    if (constraint.roleIdentifier) {
+                        return this._userService.hasRoleByIdentifier(constraint.roleIdentifier, constraint.processIdentifier);
+                    } else {
+                        return this._userService.hasRoleByName(constraint.roleName, constraint.processIdentifier);
+                    }
+                }));
+            }
+
+            if (view.access.hasOwnProperty('role')) {
+                const allowedRoles = this.parseRoleConstraints(view.access.role, url);
+
+                return  allowedRoles.some(constraint => {
+                    if (constraint.roleIdentifier) {
+                        return this._userService.hasRoleByIdentifier(constraint.roleIdentifier, constraint.processIdentifier);
+                    } else {
+                        return this._userService.hasRoleByName(constraint.roleName, constraint.processIdentifier);
+                    }
+                });
+            }
+
+            return true;
```

Comment:
should `true` be the default return value? If there are is neither a `bannedRole`, nor a `role` property, ist the `access` configured correctly, should it not just display an error for the developer, that the configuration is faulty?

Reply:
I agree. If the configuration is faulty, it should display an error.

Reply:
@minop would like to discuss this and have another idea, we will have a call tomorrow.